code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -0,0 +1,366 @@
+	if(!M) M = holder.my_atom
+	M.adjustOxyLoss(-6*REM)
+	if(M.losebreath >= 4)
+		M.losebreath -= 4
+	..()
+	return
+
+/datum/chemical_reaction/salbutamol
+	name = "Salbutamol"
+	id = "salbutamol"
+	result = "salbutamol"
+	required_reagents = list("sal_acid" = 1, "lithium" = 1, "aluminium" = 1, "bromine" = 1, "ammonia" = 1)
+	result_amount = 5
+
+datum/reagent/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	description = "Heals 25 OXY damage, but you can't talk. 33% chance of healing 1 BRUTE and 1 BURN."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolize_rate = 0.2
+
+datum/reagent/perfluorodecalin/on_mob_life(var/mob/living/carbon/human/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustOxyLoss(-25*REM)
+	M.silent = max(M.silent, 5)
+	if(prob(33))
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	..()
+	return
+
+/datum/chemical_reaction/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	result = "perfluorodecalin"
+	required_reagents = list("hydrogen" = 1, "fluorine" = 1, "oil" = 1)
+	result_amount = 3
+	required_temp = 370
+	mix_message = "The mixture rapidly turns into a dense pink liquid."
+
+datum/reagent/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	description = "Stun reduction per cycle, increases run speed slightly, minor stamina regeneration buff, stabilizes crit."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolize_rate = 0.3
+
+datum/reagent/ephedrine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.status_flags |= GOTTAGOFAST
+	if(M.losebreath >= 10)
+		M.losebreath = max(10, M.losebreath-5)
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	M.adjustStaminaLoss(-1*REM)
+	..()
+	return
+
+/datum/chemical_reaction/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	result = "ephedrine"
+	required_reagents = list("sugar" = 1, "oil" = 1, "hydrogen" = 1, "diethylamine" = 1)
+	result_amount = 4
+	mix_message = "The solution fizzes and gives off toxic fumes."
+
+datum/reagent/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	description = "Stun reduction per cycle, increases run speed slightly, minor stamina regeneration buff, stabilizes crit."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	overrides_metab = 1
+	new_metabolize_rate = 3
+datum/reagent/diphenhydramine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.drowsyness -= 1
+	M.jitteriness -= 1
+	..()
+	return
+
+/datum/chemical_reaction/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	result = "diphenhydramine"
+	required_reagents = list("oil" = 1, "carbon" = 1, "bromine" = 1, "diethylamine" = 1, "ethanol" = 1)
+	result_amount = 4
+	mix_message = "The mixture dries into a pale blue powder."
+
+datum/reagent/morphine
+	name = "Morphine"
+	id = "morphine"
+	description = "100% chance per cycle of healing 1 point each of OXY and TOX damage."
```

Comment:
You can make some morphine medipens since they're basically epipens anyway and this would be fun if this reagent jumped them out of crit temporarily too (when the reagent runs out, if they're still -100 damage, they'd fall into crit again)